### ISSUE TITLE
[BUG]: Enabling Auth giving 500 errors intermittently, Failed to update UserAttribute

### ISSUE THREAD

**Issue by dasarinaidu** [rancher/rancher#46164]
**Rancher Server Setup**
- Rancher version:  Seeing this issue on both v2.8&v2.9
 `v2.9-3c4ccdc5bc9fde3510089153b5ad58fdbe604880-head`
- Installation option (Docker install/Helm Chart): Docker and HA
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1 `v1.30.2`
- Proxy/Cert Details: Self-signed

**Describe the bug**
When i try to connect to an Auth provider, the AUTH provider connection is happening but i am seeing 500 error inbetween and the UI is showing the errors on the screen. This is happening intermittently. 

Error as below:
Failed to create or update userAttribute: failed to update UserAttribute: Operation cannot be fulfilled on userattributes.management.cattle.io "user-uuuuu": the object has been modified; please apply your changes to the latest version and try again

**To Reproduce**
1. Provision rancher
2. Login with admin
3. Users-> Users & Authentication ->Auth Provider
4. Select OpenLDAP/AzureAd
5. Provide all the details and click on Enable
6. Check for the screens and the Auth enablement 

**Result**
1. Intermittently user getting an error screen with 500 in the network call
2. Auth is enabled but it shows the error and Active status on same screen
3. Error message got disappeared after a screen refresh
4. The error is related to UserAttribute: Operation cannot be fulfilled on userattributes.management.cattle.io

**Expected Result**
1. User should not see any errors

**Screenshots**
![image](https://github.com/user-attachments/assets/ce177529-0746-42a6-9897-5b37918135e1)

**Additional context**
1. This is happening on all the auth providers
2. I m seeing this error 40% of the time

Rancher logs have the following errors:
```
2024/07/16 15:42:02 [ERROR] [AZUREAD_PROVIDER_CACHE] Failed to read the access token from Kubernetes: error getting secret cattle-global-data:azuread-access-token: secrets "azuread-access-token" not found
2024/07/16 15:42:02 [ERROR] [AZUREAD_PROVIDER_CACHE] Failed to read the access token from Kubernetes: error getting secret cattle-global-data:azuread-access-token: secrets "azuread-access-token" not found
2024/07/16 15:42:03 [ERROR] [AZUREAD_PROVIDER] Page Iterator received incorrect value of type *models.DirectoryRole
2024/07/16 15:43:15 [WARNING] Unable to determine if provider azuread was disabled, will assume that it isn't with error: error getting secret cattle-global-data:azureadconfig-applicationsecret: secrets "azureadconfig-applicationsecret" not found
2024/07/16 15:43:15 [ERROR] error syncing 'user-7hrgw': handler mgmt-auth-userattributes-controller: error refreshing user attribute user-7hrgw: error determining if Azure AD auth provider uses per-user tokens because of an error to fetch its config: error getting secret cattle-global-data:azureadconfig-applicationsecret: secrets "azureadconfig-applicationsecret" not found, requeuing
```

**Comment by dasarinaidu** [comment]
Validations completed on `v2.9-0d546ef5accf60132cbae1c170039a09a94a41b7-head` and looks good. No new issues reported.

The same code change is merged to v2.8 and v2.7. I will validate on these two envs and close this card.

**Comment by dasarinaidu** [comment]
Validated this on v2.8 & v2.7 also, I did not see any issues and closing this card.
